FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/CustomFacetGatewayTest.php

```diff
@@ -0,0 +1,416 @@
+            ['isSearchPage', true],
+            ['isCategoryPage', false],
+            ['findologicDI', false]
+        ];
+
+        // Create mock object for Shopware Session and explicitly return the values
+        $mockSession = $this->getMockBuilder('\Enlight_Components_Session_Namespace')
+            ->setMethods(['offsetGet'])
+            ->getMock();
+        $mockSession->method('offsetGet')
+            ->willReturnMap($sessionArray);
+
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('session', $mockSession);
+
+        $mockOriginalService = $this->getMockBuilder(
+            '\Shopware\Bundle\StoreFrontBundle\Gateway\DBAL\CustomFacetGateway'
+        )
+            ->setMethods(['getList'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $mockOriginalService->expects($this->never())->method('getList');
+
+        $mockUrlBuilder = $this->getMockBuilder('\FinSearchUnified\Helper\UrlBuilder')->getMock();
+        $mockUrlBuilder->expects($this->once())->method('setCustomerGroup');
+        $xmlResponse = new SimpleXMLElement('<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>');
+
+        $results = $xmlResponse->addChild('results');
+        $results->addChild('count', 2);
+        $filters = $xmlResponse->addChild('filters');
+        foreach ($filterData as $data) {
+            $filter = $filters->addChild('filter');
+            foreach ($data as $key => $value) {
+                $filter->addChild($key, $value);
+            }
+        }
+        $mockUrlBuilder->expects($this->once())
+            ->method('buildCompleteFilterList')
+            ->willReturn(new \Zend_Http_Response(200, [], $xmlResponse->asXML()));
+
+        $originalHydrator = Shopware()->Container()->get('fin_search_unified.custom_listing_hydrator');
+
+        $mockHydrator = $this->getMockBuilder(
+            '\FinSearchUnified\Bundle\StoreFrontBundle\Gateway\Findologic\Hydrator\CustomListingHydrator'
+        )
+            ->setMethods([])
+            ->getMock();
+        $facetArray = [];
+        foreach ($xmlResponse->filters->filter as $filter) {
+            $facetArray[] = $originalHydrator->hydrateFacet($filter);
+        }
+        $mockHydrator->expects($this->exactly(count($filterData)))
+            ->method('hydrateFacet')
+            ->willReturnOnConsecutiveCalls($facetArray);
+
+        $facetGateway = new CustomFacetGateway(
+            $mockOriginalService,
+            $mockHydrator,
+            $mockUrlBuilder
+        );
+
+        $customFacets = $facetGateway->getList([3], $context);
+
+        /** @var CustomFacet $customFacet */
+        foreach ($customFacets[0] as $key => $customFacet) {
+            $this->assertSame(
+                $facetData[$key]['name'],
+                $customFacet->getName(),
+                sprintf("Expected custom facet's name to be %s", $facetData[$key]['name'])
```

Comment:
I like the way you use sprintf :wink:

Reply:
sarcasm? :stuck_out_tongue_closed_eyes:

Reply:
Nope. Would have written that line the same way :smile: